cmd/api/src/config/reflect.go

```diff
@@ -253,8 +253,9 @@ func SetValue(target any, path, value string) error {
 		cursor    = indirectOf(target)
 	)
 
-	for idx, nextPathPart := range pathParts {
+	for idx := 0; idx < len(pathParts); idx++ {
```

Comment:
The changes to this loop and associated logic allows us to parse field names that contain underscores while still using underscores as path separators. Previous logic did allow for underscores in field names, but only when it was the last field in a path. Fields with underscores at the beginning or middle of a path would be ignored previously.